packages/stepper/src/Stepper.spec.tsx

```diff
@@ -0,0 +1,272 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import type { RenderResult } from '@testing-library/react';
+import { Step, Stepper } from '.';
+
+function isVisible(element: HTMLElement): boolean {
+  try {
+    expect(element).toBeVisible();
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+function onlyElement(elements: Array<HTMLElement>): HTMLElement | null {
+  expect(elements.length).toBeLessThanOrEqual(1);
+  return elements[0] ?? null;
+}
+
+function renderSteps(
+  {
+    allSteps,
+    currentStep,
+    maxDisplayedSteps,
+  }: {
+    allSteps: Array<string>;
+    currentStep: number;
+    maxDisplayedSteps: number;
+    expectedSteps?: Array<string>;
+  },
+  renderFunction: typeof render | RenderResult['rerender'],
+) {
+  const result = renderFunction(
+    <Stepper currentStep={currentStep} maxDisplayedSteps={maxDisplayedSteps}>
+      {allSteps.map(step => (
+        <Step key={step}>{step}</Step>
+      ))}
+    </Stepper>,
+  );
+  return result ? result.rerender : undefined;
+}
+
+function assertVisibleSteps({
+  allSteps,
+  currentStep,
+  expectedSteps = allSteps,
+}: {
+  allSteps: Array<string>;
+  currentStep?: number;
+  expectedSteps?: Array<string>;
+}) {
+  const visiblePreviousStepsElement = onlyElement(
+    screen.queryAllByLabelText('Previous steps').filter(isVisible),
+  );
+
+  if (expectedSteps.includes('Previous steps')) {
+    expect(visiblePreviousStepsElement).not.toBeNull();
+  } else {
+    expect(visiblePreviousStepsElement).toBeNull();
```

Comment:
Generally I prefer `toBeInDocument` as the error messages when it's incorrect is more rich. Can we use that instead?